### PR TITLE
refactor DRY remote docs, Graph Client with Mermaid remote docs

### PR DESCRIPTION
**Pull request by dimaMachina**
fixes https://github.com/graphprotocol/docs/pull/379#discussion_r1220374947

### CODE REVIEW

File: website/scripts/fetch-remote-filepaths.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      'firehose-setup/near/README.md',
+      'firehose-setup/cosmos/single-machine-deployment.md',
+      'integrate-new-chains/README.md',
+      'references/README.md',
+      'setup/cosmos/README.md',
+      'setup/ethereum/README.md',
+      'setup/ethereum/reprocessing-history.md',
+      'setup/ethereum/synchronization.md',
+      'concepts/README.md',
+    ].includes(filePath),
```
Comment (B2o5T) [inline website/scripts/fetch-remote-filepaths.ts:78]: @saihaj could you write in the comments for each file why you exclude them as I did in 
https://github.com/graphprotocol/docs/blob/refactor-remote/website/scripts/fetch-remote-filepaths.ts#L51-L55

File: website/scripts/fetch-remote-filepaths.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+type Params = {
+  user: string
+  repo: string
+  branch: string
+  docsPath: string
+  outputPath: string
+  filterDocs?: (filePath: string) => boolean
+}
+
+async function fetchRemoteFilePaths({ user, repo, branch, docsPath, outputPath, filterDocs }: Params): Promise<void> {
```
Comment (B2o5T) [inline website/scripts/fetch-remote-filepaths.ts:15]: @benface this fn will avoid GitHub rate limit :)

File: website/pages/en/firehose/[[...slug]].mdx [hunk 3]
```diff
@@ -38,45 +33,27 @@ export async function getStaticProps({ params: { slug = ['README'] } }) {
...
-            if (node.url.startsWith('/')) {
-              // (foo)[/foo/bar]
-              node.url = node.url.replace('/', '/firehose/')
-            } else if (!node.url.includes('/') && node.url.endsWith('.md')) {
-              // (foo)[foo.md]
-              node.url = [...slug.slice(0, -1), node.url].join('/')
-            }
-          })
-          done()
-        },
```
Comment (B2o5T) [inline website/pages/en/firehose/[[...slug]].mdx:63]: @saihaj This seems an old solution that was refactored by @benface

I extracted function from substreams docs to reuse here, but some links are broken, could you take a look why? maybe due to excluding files that you made?

File: website/pages/en/_meta.js [hunk 4]
```diff
@@ -10,6 +10,7 @@ export default {
   glossary: '',
   'arbitrum-faq': '',
   tokenomics: 'Tokenomics',
+  'graph-client': 'Graph Client',
```
Comment (B2o5T) [inline website/pages/en/_meta.js:13]: does it the right place in the sidebar for Graph Client docs?

Comment (saihaj) [reply website/pages/en/_meta.js:13]: I think this should be part of Querying section

File: website/pages/en/firehose/[[...slug]].mdx (hunk 3 above)
Comment (saihaj) [reply website/pages/en/firehose/[[...slug]].mdx:63]: https://github.com/graphprotocol/docs/pull/399/commits/cf5b96f4ca282be2fbaaccca57b47599c5c003cc

File: website/scripts/fetch-remote-filepaths.ts (hunk 2 above)
Comment (benface) [reply website/scripts/fetch-remote-filepaths.ts:13]: Nice!

**Review by benface — APPROVED** [review]
Very nice.

File: website/pages/en/_meta.js (hunk 4 above)
Comment (B2o5T) [reply website/pages/en/_meta.js]: moved!

**Pull request merged**